### ISSUE TITLE
Upgrade TFKerasPruningCallback to work with TF 2.3

### ISSUE THREAD

**Issue by brethvoice** [optuna/optuna#1936]
## Description

According to the following link, "This callback is intend to be compatible for TensorFlow v1 and v2, but only tested with TensorFlow v1."

https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.TFKerasPruningCallback.html#optuna.integration.TFKerasPruningCallback

## Motivation

TF 2.3 is working well and worthy of recognition.  Currently, the hook is not working to prune trials.

## Description

Make the hook work with TF 2.3+.

## Alternatives (optional)

Fork the code and make it work myself.

## Additional context (optional)

For an example of how it does not work, run 'study_mnist_classifier_settings.py` at the following repo:
[https://github.com/brethvoice/optuna_demo_MNIST](https://github.com/brethvoice/optuna_demo_MNIST)

It will take about 30 minutes to run but you can observed how no trials are *ever* pruned.  Max trials = 10, max timeout is 30 minutes.

**Comment by brethvoice** [comment]
Possibly need to approve/reject this pull request first?
https://github.com/optuna/optuna/pull/1184

**Comment by toshihikoyanase** [comment]
@brethvoice Thank you for your report. I don't think this issue is directly related to #1184, and we can work on each issue independently.

@sfujiwara I thought `TFKerasPruningCallback` worked with the latest `tensorflow`. Do you have any comments on this issue?

**Comment by sfujiwara** [comment]
> According to the following link, "This callback is intend to be compatible for TensorFlow v1 and v2, but only tested with TensorFlow v1."
> https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.TFKerasPruningCallback.html#optuna.integration.TFKerasPruningCallback

I guess the above statement is out dated and we have to fix it. When I implemented `TFKerasPruningCallback`, We tested the callback on v1 because TensorFlow v2 was experimental. But currently we use TensorFlow v2 on testing as below (currently `tensorflow` without specifying version would install v2):
https://github.com/optuna/optuna/blob/1375de019a824a705fa22682491385350ff283b1/setup.py#L113-L138

**Comment by sfujiwara** [comment]
@brethvoice I tried your code. It's a little bit complicated to understand perfectly, but I found the metric name is wrong:

https://github.com/brethvoice/optuna_demo_MNIST/blob/db5544d919840a12647107700254bc60f2353257/study_mnist_classifier_settings.py#L114

It seems that you use `val_categorical_accuracy` in your training code, but `TFKerasPruningCallback` watch `val_acc`.

**Comment by sfujiwara** [comment]
When we set non-existing metric name as `self._monitor` in `TFKerasPruningCallback`, it is ignored and `current_score` is `None` as below:
https://github.com/optuna/optuna/blob/1375de019a824a705fa22682491385350ff283b1/optuna/integration/tfkeras.py#L42-L56

@toshihikoyanase I think it would be user friendly to warn in such case. What do you think about it?

**Comment by toshihikoyanase** [comment]
@sfujiwara Thank you for your investigation. I agree with you.
Actually, we have tracked the implementation of such warnings in #1226, and @bigbird555 kindly added the warning to `KerasPruningCallback` in #1759.

**Comment by brethvoice** [comment]
> @brethvoice I tried your code. It's a little bit complicated to understand perfectly, but I found the metric name is wrong:
> 
> https://github.com/brethvoice/optuna_demo_MNIST/blob/db5544d919840a12647107700254bc60f2353257/study_mnist_classifier_settings.py#L114
> 
> It seems that you use `val_categorical_accuracy` in your training code, but `TFKerasPruningCallback` watch `val_acc`.

That worked!  I just had to use `val_categorical_accuracy` and pruning is happening now.  Thank you @sfujiwara!!

**Comment by brethvoice** [comment]
Currently I am using the following:
```
keras_pruner = optuna.integration.TFKerasPruningCallback(
    trial,
    'val_categorical_accuracy',
)
sampler_multivariate = optuna.samplers.TPESampler(multivariate=True)
study = optuna.create_study(
    sampler=sampler_multivariate,
    direction='maximize',
    pruner=optuna.pruners.PercentilePruner(
        percentile=PERCENTILE_FOR_PRUNING,
        n_startup_trials=NUMBER_OF_TRIALS_BEFORE_PRUNING,
        n_warmup_steps=WARMUP_EPOCHS_BEFORE_PRUNING,
    ),
)
study.optimize(
    objective,
    n_trials=MAXIMUM_NUMBER_OF_TRIALS_TO_RUN,
    timeout=MAXIMUM_SECONDS_TO_CONTINUE_STUDY,
    gc_after_trial=True,
    )
```
and it is working.  The relevant Tensorflow code I have is:
```
classifier_model.fit(
    standard_object.train_split_images,
    standard_object.train_split_labels,
    epochs=MAXIMUM_EPOCHS_TO_TRAIN,
    validation_data=standard_object.validate_split_data,
    verbose=VERBOSITY_LEVEL_FOR_TENSORFLOW,
    callbacks=standard_object.callbacks,
    batch_size=standard_object.batch_size,
)
```
from [this](https://github.com/brethvoice/optuna_demo_MNIST/commit/ddec83dd32f9cb2ee70c8ac7398db6e53c0549f8) commit.
